includes/class-revision-strike.php

```diff
@@ -185,33 +187,133 @@ public function strike( $args = array() ) {
 	 *                         we can purge the post revisions.
 	 * @param int   $limit     The maximum number of revision IDs to retrieve.
 	 * @param array $post_type The post types for which revisions should be located.
+	 * @param int   $keep      Keep at least this number of revisions for a post, regardless of age.
 	 *
-	 * @return array An array of post IDs (unless 'fields' is manipulated in $args).
+	 * @return array An array of revision IDs
 	 */
-	protected function get_revision_ids( $days, $limit, $post_type ) {
-		global $wpdb;
+	protected function get_revision_ids( $days, $limit, $post_type, $keep ) {
 
 		// Return early if we don't have any eligible post types.
-		if ( ! $post_type ) {
+		if ( empty( $post_type ) ) {
 			return array();
 		}
 
 		$post_type    = array_map( 'trim', explode( ',', $post_type ) );
-		$revision_ids = $wpdb->get_col( $wpdb->prepare(
+
+		// Get a list of post IDs and their revisions.
+		$results = $this->query_post_and_revision_ids( $post_type, $days );
+
+		// Allow filtering of the query results.
+		$posts = apply_filters( 'revisionstrike_revisions_query_results', $results );
```

Comment:
This PR is introducing quite a bit of new code, but we want to be sure that new filters are being added strategically. At the very least, these need full documentation, but I'd say until there's cause for it maybe we don't overload the plugin with filters right away.